src/main/java/de/tum/cit/aet/artemis/iris/web/open/PublicPyrisStatusUpdateResource.java

```diff
@@ -171,7 +183,58 @@ public ResponseEntity<Void> setStatusOfIngestionJob(@PathVariable String runId,
         }
 
         pyrisStatusUpdateService.handleStatusUpdate(ingestionWebhookJob, statusUpdateDTO);
-
         return ResponseEntity.ok().build();
     }
+
+    /**
+     * Retrieves the overall ingestion state of a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of all lectures in a course, identified by its `lectureId`. The ingestion state can be aggregated from
+     * multiple lecture units or can reflect the overall status of the lecture ingestion process.
+     * </p>
+     *
+     * @param courseId the ID of the lecture for which the ingestion state is being requested
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture,
+     */
+    @GetMapping("courses/{courseId}/lectures/ingestion-state")
+    @EnforceNothing
+    public ResponseEntity<Map<Long, IngestionState>> getStatusOfLectureIngestion(@PathVariable long courseId) {
+        try {
+            return ResponseEntity.ok(pyrisConnectorService.getLecturesIngestionState(courseId));
+        }
+        catch (Exception e) {
+            log.error("Error fetching ingestion state for course {}", courseId, e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
```

Comment:
This is not a good way of error handling. I suggest you either suppress the error completely or you provide a reason to the client why the request failed. I prefer the second option, because this would make it easier to identify errors.